hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMORDataSource.scala

```diff
@@ -516,17 +516,14 @@ class TestMORDataSource extends HoodieClientTestBase {
     checkAnswer((1, "a0", 12, 101, false))
 
     writeData((1, "a0", 16, 97, true))
-    // Ordering value will not be honored for a delete record as the payload is sent as empty payload
-    checkAnswer((1, "a0", 16, 97, true))
+    // Ordering value will be honored, the delete record is considered as obsolete
+    // because it has smaller version number (97 < 101)
+    checkAnswer((1, "a0", 12, 101, false))
 
     writeData((1, "a0", 18, 96, false))
-    // Ideally, once a record is deleted, preCombine does not kick. So, any new record will be considered valid ignoring
-    // ordering val. But what happens ini hudi is, all records in log files are reconciled and then merged with base
```

Comment:
What this test describes seems plainly an issue to me. 
Let's 
 - Create a ticket to address it 
 - Keep the test flow (either w/in this test or separately) to make sure we have a reference point when we get to fix it

Reply:
This test describes is just so weird because it tries to explain why the error/unexpected case happens, i don't think there is any necessity to keep it.

If you think the fix the hard to understand, let's make it clear in this PR.

Reply:
So what is the meaning to keep an explanation for an incorrect case ? Let's just fix that and the confusion is gone

Reply:
@alexeykudinkin can you clarify the high level concern here? do you think the test is wrong (with the changes in the PR) or we need more cases or you unearthed some bug we still need to fix in the PR